sdk/servicebus/azure-servicebus/tests/test_repr.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import uamqp
+from azure.servicebus import ServiceBusMessage, ServiceBusReceivedMessage
+from azure.servicebus._common.constants import (
+    _X_OPT_PARTITION_KEY,
+    _X_OPT_VIA_PARTITION_KEY,
+    _X_OPT_SCHEDULED_ENQUEUE_TIME
+)
+
+def test_servicebus_message_repr():
+    message = ServiceBusMessage("hello")
+    assert "application_properties=None, session_id=None," in message.__repr__()
+    assert "content_type=None, correlation_id=None, to=None, reply_to=None, reply_to_session_id=None, subject=None, scheduled_enqueue_time_utc" in message.__repr__()
+
+def test_servicebus_received_message_repr():
+    uamqp_received_message = uamqp.message.Message(
+        body=b'data',
+        annotations={
+            _X_OPT_PARTITION_KEY: b'r_key',
+            _X_OPT_VIA_PARTITION_KEY: b'r_via_key',
+            _X_OPT_SCHEDULED_ENQUEUE_TIME: 123424566,
+        },
+        properties=uamqp.message.MessageProperties()
+    )
+    received_message = ServiceBusReceivedMessage(uamqp_received_message, receiver=None)
+    assert "application_properties=None, session_id=None" in received_message.__repr__()
+    assert "content_type=None, correlation_id=None, to=None, reply_to=None, reply_to_session_id=None, subject=None, scheduled_enqueue_time_utc" in received_message.__repr__()
```

Comment:
I think we also need to verify in the assertion that `__repr__` contains the `annotations`.
```
        annotations={
            _X_OPT_PARTITION_KEY: b'r_key',
            _X_OPT_VIA_PARTITION_KEY: b'r_via_key',
            _X_OPT_SCHEDULED_ENQUEUE_TIME: 123424566,
        },
```

Reply:
done

Reply:
I need to apologize that it's my fault for not making it clear -- I see you exposing the `annotations` in the `__repr__`.
the `annotations` is not a public ivar/property, so I think we should not expose it.

I intended to check those values within the `annotations` are included in the repr as each of them you set is related to a public property.
like:
```
assert "partition_key=r_key, scheduled_enqueue_time_utc=123424566" in received_message.__repr__()
```

----

`via_partition_key` is not a public property on `ServiceBusMessage` by design as it's part of the transaction feature which we are missing in current uamqp.